packages/auth/src/CurrentUserInitializer.php

```diff
@@ -8,15 +8,16 @@
 use Tempest\Container\DynamicInitializer;
 use Tempest\Container\Tag;
 use Tempest\Reflection\ClassReflector;
+use UnitEnum;
 
 final readonly class CurrentUserInitializer implements DynamicInitializer
 {
-    public function canInitialize(ClassReflector $class, ?string $tag): bool
+    public function canInitialize(ClassReflector $class, null|string|UnitEnum $tag): bool
```

Comment:
While working on this PR, I discovered that we didn't support enums as tags in large parts of the container. That's why I made multiple changes throughout the codebase to properly support them